File: zapit-frontend/src/app/invoice/invoice.ts
import { Merchant } from '../merchant/merchant';
import { User } from '../user/user';
import { InvoiceItem } from './invoice-item';

export interface Invoice {
  id?: string;
  identifier?: string;
  invoiceItems?: InvoiceItem[];
  salesTax: number;
  additionalCharges: number;
  total: number;
  eligiblePoints?: number;
  status?: string;
  issuedBy?: Merchant;
  claimedBy?: User;
  claimedOn?: number;
  createdOn?: number;
  updatedOn?: number;
}

File: zapit-backend/src/main/java/com/shaunwah/zapitbackend/model/Invoice.java
package com.shaunwah.zapitbackend.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.sql.Timestamp;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Invoice {
    private String id;
    private String identifier;
    private List<InvoiceItem> invoiceItems;
    private Double salesTax;
    private Double additionalCharges;
    private Double total;
    private Double eligiblePoints;
    private String status;
    private Merchant issuedBy;
    private User claimedBy;
    private Long claimedOn;
    private Long createdOn;
    private Long updatedOn;
}
